frontend/src/features/comparisonSeries/ComparisonSeries.tsx

```diff
@@ -255,7 +255,19 @@ const ComparisonSeries = ({
   }
 
   if (redirectTo && step >= length) {
-    return <Redirect to={{ pathname: redirectTo }} />;
+    const keepUIDs = searchParams.get('keep_uids_after_redirect');
+    const futureSearchParams = new URLSearchParams();
+
+    if (keepUIDs === 'true') {
+      futureSearchParams.append('uidA', uidA);
+      futureSearchParams.append('uidB', uidB);
+    }
+
+    return (
+      <Redirect
+        to={{ pathname: redirectTo, search: futureSearchParams.toString() }}
+      />
+    );
```

Comment:
This code triggers the infinite reload of the entity selectors.

When a user starts the tutorial from the comparison UI of the home page, the `keep_uids_after_redirect` is set. After the last tutorial comparison the user is redirected to the regular comparison component. The presence of the two UIDs in the URL triggers the problematic behaviour.

Why do we need `keep_uids_after_redirect` here?

The purpose of `keep_uids_after_redirect` is to address a specific situation. Usually after the tutorial the users are redirected to an empty regular comparison UI. This is problematic for anonymous users having already made more than 4 comparisons. When they click on the compare the videos button on the home page, they can log in, and are redirected to the comparison series which then redirect the users to the empty regular comparison UI*. The two UIDs of the home page are not preserved in this final redirection. To simplify, for thiis kind of user, clicking on compare the videos on the home page redirects to an empty comparison.

Even if the mechanism of `keep_uids_after_redirect` can be simplified to avoid the infinite reloading problem, it does not seem to be the root cause of the issue. If an entity selector reloads videos again and again, it may reveal that several "things" are trying to update it. According to the errors in the console, it may be due to unmounted components with not properly cleaned effects.

Reply:
This is what we can read in the console after submitting the last comparison of the tutorial.

![capture](https://user-images.githubusercontent.com/39056254/196181223-ddf2608c-2383-49ae-a8fa-f52a4c5ccca9.png)

Reply:
After more investigation, it may be related to the last call of `afterSubmitCallback`. This last call is made after the last click on the submit button, at the same time the `ComparisonSeries` is redirecting the user.

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/1128/commits/a5afc40869a09ffbd82f8fa5f06d4c93a8006f3d